### ISSUE TITLE
Some ideas

### ISSUE THREAD

**Issue by potatoqualitee** [marcingminski/sqlwatch#4]
Awesome project! Looking forward to seeing more.

Something to consider - making a PowerShell module that helps with the installation. You can include Invoke-SqlCmd2.ps1 in an `internal\functions` directory, then use it to execute commands without an SMO dependency.

The one thing that's 🤔 is which prefix to use (to avoid clashing with microsoft). I'm thinking SW and then aliasing two commands 
`Install-SWatch` -> `Install-SqlWatch`
`Uninstall-SqlWatch` -> `Uninstall-SqlWatch`

Then you can make a command like
Install-SWatch -SqlInstance sql01 -Database DBA
Uninstall-SqlWatch
Reset-SWDatabase -SqlInstance sql01 -Database DBA
Read-SWData -SqlInstance sql01 -Database DBA

If you set a dependency to PSFramework, you can write the app config locally then do like
Get-SWConfig
Set-SWConfig
Which writes the default sqlinstance and db, so users can just execute

Read-SWData

Without -SqlInstance sql01 -Database DBA and it'll still work.

In the future, if you'd be interested, we can host you at /sqlcollaborative - you'd have admin privs to your repo but have greater exposure.

**Comment by leedsdata** [comment]
Happy to pick this up unless you've already started on it?

**Comment by marcingminski** [comment]
Hi, 
@potatoqualitee thank you for all the tips and the kind words :) 
@LeedsDBA I have not started yet but I would like to consider two things:

1. I would want to retain the ability to install from a SQL file for those who don't know PS. I know personally a number of people who would be interested in the data logger but do not do PS. 

1. we are looking at potential migrations if we ever upgrade schema and have to handle previous versions, this is something that would have to be handled on per-case basis in the upgrade/install file.

With this in mind, wouldn't the easiest way be to execute install.sql and uninstall.sql via PS. I guess it would be similar to how Olla Hallengren's scripts are being installed via dbatools? I have not looked at it.

I'm open for suggestions :) Glad to see the project is gaining interest :)

**Comment by leedsdata** [comment]
Hi @marcingminski 

=1= 
For setup, in my opinion i think Powershell *should* make it easier, but i understand the apprehension. I think it would be great to install the below, one liner - done :)

Install-SWatch -SqlInstance AppUAT01

Like you mention this could just be a wrapper for a install.sql

But if you did look to manage the whole install via PowerShell it also allows you to split your install files up into smaller units of code for maintenance (your installer could be responsible for deploying the files in a setup folder for example). 

The project will no doubt get bigger with more tables and more things you might want to collect, i think that might be the easiest way to provide both the end-user with the ability to deploy/upgrade their version and for you to work with a growing project in terms of files.

=2= 
setup.sql at the moment checks to see if an object doesn't exist, if so, then deploy object. This wouldn't allow your end-users to upgrade v1 to v2 if you make changes to those source tables for example. It might be better to use a state-based SSDT approach and let PowerShell do the install/upgrade?

Another option is to provide both options to the end-user - a PowerShell installer which deals with multiple setup files, and an install.sql which could be compiled dynamically on build. Ticks both boxes :)

Interested to hear other points of view! :)

**Comment by marcingminski** [comment]
Yeah if I make a change to the schema I'd have to handle it in the setup script too. Pain. I would normally use VS project and sqlpackage.exe triggered from a PowerShell - should we look at this as an option and move to VS project?

**Comment by larspl** [comment]
Awesome Project Marcin, i just figured out some ideas in the past. i like to support this project.

**Comment by marcingminski** [comment]
Thank you sir!

**Comment by marcingminski** [comment]
I will have to close this thread as it prevents me from deleting an old branch. Thank you for all the valuable comments!

**Comment by marcingminski** [comment]
Just going through old "tickets". @potatoqualitee can you expand on the `Get-SWConfig` and `
Set-SWConfig` what the benefit would be etc, please? I am not familiar with these commands but sounds like something that might be useful to be able to write deployment manifests?
